correlation/src/main/java/com/networknt/correlation/CorrelationHandler.java

```diff
@@ -59,21 +67,42 @@ public CorrelationHandler() {
     public void handleRequest(final HttpServerExchange exchange) throws Exception {
         // check if the cid is in the request header
         String cId = exchange.getRequestHeaders().getFirst(HttpStringConstants.CORRELATION_ID);
+        
+        //Fetching serviceId to set in MDC
+        Map<String, Object> serverConfig = Config.getInstance().getJsonMapConfigNoCache(SERVER_CONFIG_NAME);
+        if (serverConfig != null) {
+        	serviceId = (String) serverConfig.get(SERVICEID_KEY);
+        }
+        
         if(cId == null) {
         	// if not set, check the autgen flag and generate if set to true
         	if(config.isAutogenCorrelationID()) {
 	            // generate a UUID and put it into the request header
 	            cId = Util.getUUID();
 	            exchange.getRequestHeaders().put(HttpStringConstants.CORRELATION_ID, cId);
 	            String tId = exchange.getRequestHeaders().getFirst(HttpStringConstants.TRACEABILITY_ID);
+	            
+	            MDC.put(CID, cId);
+	            if(!"".equalsIgnoreCase(serviceId)) {
```

Comment:
This part I'm not sure, @stevehu  should we put service id in the correlation handler? it seems mix the feature, but will also be too much to create a module just for putting the service id?